aiidalab_eln/eln/__init__.py

```diff
@@ -0,0 +1,28 @@
+import traitlets
+import ipywidgets as ipw
+
+class ElnConnector(ipw.VBox):
+
+    eln_instance = traitlets.Unicode()
+    eln_type = traitlets.Unicode()
+
+    def __init__(self, **kwargs):
+        """Connect to an ELN
+
+        Args:
+            eln_instance (str): URL which points to the eln instance.
+            user (str): ELN user.
+            token (str): ELN access token (or user password).
+            eln_type (str): ELN type, e.g. "cheminfopy"
+        """
+        super().__init__(**kwargs)
+    
+    def connect(self):
+        raise NotImplementedError(f"{self.__class__.__name__} does not implement the 'connect' method")
+    
+    def is_connected(self):
```

Comment:
- if you have per-sample authentication you'll also need to pass sample information. 
- i'm still not sure if the `is_connected` is the full abstraction as you might have different levels of connections (only read allowed: might also be the case in openbis for some entries), read+write allowed. That is, after a simple "connection" check, a request might still fail due to lack of rights. 

The alternative would be to have this `is_connected` call and then the `has_rights` check before you actually do an operation. 

In `cheminfopy` this currently, for example a `sample.has_right(self, right: str) -> bool` call. (I'll add it also on the user level for sample/reaction).

Reply:
I agree with you.

Reply:
we had a separate discussion on this topic and agreed on the following:

* `is_connected` function will be kept. In the context of cheminfo it will check whether the read, write and addAttachment rights. If this check fails, the user will be redirected to the eln configuration page.
* In case the previous check is OK, then the attempt to send the information to ELN is performed. In case of no rights, a specific error will be raised. This code will catch the error and will inform the user that they have no rights.
* Finally, in case of any other error, an `Unknown` error will be raised.

@kjappelbaum please confirm that the description above is correct.